ip-range-matcher/src/com/dataiku/dip/plugins/iprangematcher/IPRangeMatcher.java

```diff
@@ -0,0 +1,278 @@
+        }
+
+        @Override
+        public IPRangeMatcher build(Parameter parameter) throws Exception {
+            return new IPRangeMatcher(parameter);
+        }
+
+    };
+
+
+    public IPRangeMatcher(Parameter parameter) {
+        this.parameter = parameter;
+    }
+
+    @Override
+    public void init() throws Exception {
+        ipColumn = cf.column(parameter.ipColumn);
+
+        if (parameter.action == DetectorProcessorAction.CREATE_FLAG_BOOLEAN || parameter.action == DetectorProcessorAction.CREATE_FLAG_MATCH) {
+            if (StringUtils.isBlank(parameter.flagColumn)) {
+                throw ErrorContext.iaef("Missing 'flag column' parameter");
+            }
+
+            flagColumn = cf.columnAfter(parameter.ipColumn, parameter.flagColumn);
+        }
+
+        String ranges = parameter.ranges;
+
+        rangeList = new ArrayList<>();
+
+        if (StringUtils.isBlank(parameter.ranges)) {
+            return;
+        }
+
+        try (Scanner scanner = new Scanner(ranges)) {
+            Matcher rangeMatcher = RANGE_PATTERN.matcher("");
+            Matcher cidrMatcher = CIDR_PATTERN.matcher("");
+
+            String line;
+            while (scanner.hasNextLine()) {
+                line = scanner.nextLine();
+
+                rangeMatcher.reset(line);
+                cidrMatcher.reset(line);
+
+                if (rangeMatcher.find()) {
+                    int rangeStart = ipToInteger(rangeMatcher.group(1), rangeMatcher.group(2), rangeMatcher.group(3), rangeMatcher.group(4));
+                    int rangeEnd = ipToInteger(rangeMatcher.group(5), rangeMatcher.group(6), rangeMatcher.group(7), rangeMatcher.group(8));
+
+                    rangeList.add(new Range(rangeStart, rangeEnd, line));
+                } else if (cidrMatcher.find()) {
+                    int addr = ipToInteger(cidrMatcher.group(1), cidrMatcher.group(2), cidrMatcher.group(3), cidrMatcher.group(4));
+                    int mask = (-1) << (32 - Integer.parseInt(cidrMatcher.group(5)));
+
+                    int rangeStart = addr & mask;
+                    int rangeEnd = rangeStart + (~mask);
+
+                    rangeList.add(new Range(rangeStart, rangeEnd, line));
+                }
+            }
+        }
+
+        Collections.sort(rangeList, new Comparator<Range>() {
+            @Override
+            public int compare(Range r1, Range r2) {
+                if (r1.getStart() == r2.getStart()) {
+                    return r1.getEnd() - r2.getEnd();
+                } else {
+                    return r1.getStart() - r2.getStart();
+                }
+            }
+        });
+
+
+    }
+
+    private int ipToInteger(String a, String b, String c, String d) {
+        return ipToInteger(Integer.parseInt(a),
+                b == null ? 0 : Integer.parseInt(b),
+                c == null ? 0 : Integer.parseInt(c),
+                d == null ? 0 : Integer.parseInt(d));
```

Comment:
`Integer.parseInt` calls should be replaced by a private static method that does `parseInt` + check 0-255 boundaries.